staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go

```diff
@@ -58,35 +90,61 @@ func NewFromInterface(subjectAccessReview authorizationclient.SubjectAccessRevie
 }
 
 // New creates a new WebhookAuthorizer from the provided kubeconfig file.
+// authorized and unauthorized TTL are durations for which the webhook
+// authorized verdicts are cached, for "yes" and "no" verdicts respectively.
+// resolver knows how to map a Kubernetes service name
+// ("servicename.namespace.svc") into an endpoint IP address. proxyTransport
+// knows how to dial out to in-cluster endpoints.
 //
-// The config's cluster field is used to refer to the remote service, user refers to the returned authorizer.
+// The config's cluster field is used to refer to the remote service, user
+// refers to the returned authorizer.
 //
 //     # clusters refers to the remote service.
 //     clusters:
 //     - name: name-of-remote-authz-service
 //       cluster:
-//         certificate-authority: /path/to/ca.pem      # CA for verifying the remote service.
-//         server: https://authz.example.com/authorize # URL of remote service to query. Must use 'https'.
+//         # CA for verifying the remote service.
+//         certificate-authority: /path/to/ca.pem
+//         # URL of remote service to query.  Must use 'https'.
+//         server: https://authz.example.com/authorize
 //
 //     # users refers to the API server's webhook configuration.
 //     users:
 //     - name: name-of-api-server
 //       user:
-//         client-certificate: /path/to/cert.pem # cert for the webhook plugin to use
-//         client-key: /path/to/key.pem          # key matching the cert
-//
-// For additional HTTP configuration, refer to the kubeconfig documentation
-// https://kubernetes.io/docs/user-guide/kubeconfig-file/.
-func New(kubeConfigFile string, authorizedTTL, unauthorizedTTL time.Duration) (*WebhookAuthorizer, error) {
-	subjectAccessReview, err := subjectAccessReviewInterfaceFromKubeconfig(kubeConfigFile)
+//         # Certificate for the webhook plugin to use when calling out to the
+//         # webhook server.
+//         client-certificate: /path/to/cert.pem
+//         # The client's private key that signed the certificate above.
+//         client-key: /path/to/key.pem
+func New(
```

Comment:
the webhook authorizer is used by multiple clients, other than just the main kube-apiserver (which is why it lives under `k8s.io/apiserver`, not `k8s.io/kubeapiserver`).

only kube-apiserver can be expected to have the ability, or the need (and only on some deployment topologies) to do manual resolution of service names to IPs.

there is already a `NewFromInterface` constructor that allows passing in a client that submits subjectaccessreviews opaquely. If kube-apiserver wanted to use non-dns resolution to dial service IPs, it should construct a client to do that and use that constructor, rather than make the webhook authorizer aware of service resolution like this

Reply:
Thanks for the comments.  I'll wait a bit to give anyone else opportunity to chime in.  Then will go implement this change.